[Lens] Multiple y axes

flash1293 opened the pull request:
Fixes https://github.com/elastic/kibana/issues/53663
Fixes https://github.com/elastic/kibana/issues/68260

This PR adds the ability to configure two separate y axes in XY charts.

<img width="569" alt="Screenshot 2020-06-26 at 14 32 11" src="https://user-images.githubusercontent.com/1508364/85857388-0db53180-b7ba-11ea-8699-ce17e18dfdfb.png">
<img width="1063" alt="Screenshot 2020-06-26 at 14 34 16" src="https://user-images.githubusercontent.com/1508364/85857427-2291c500-b7ba-11ea-986e-b29c13a1a799.png">

These changes are limited to the xy visualization, expect for one small fix in `x-pack/plugins/lens/public/editor_frame_service/editor_frame/config_panel/layer_panel.tsx` - there is a flag `enableDimensionEditor` on the dimension group whether to show a second visualization-controlled tab in the dimension editor, but it wasn't used to actually determine whether to show the editor or not.


## Axis assignment

The process of assigning metric to axes follows this algorithm:
* Split up all metrics by their type (auto/left/right) - this results in three lists
* Move all "auto" metrics to either left or right:
  * If a metrics formatter is compatible with all formatters of other left axis metrics, it becomes a left axis metric
  * If a metrics formatter is compatible with all formatters of other right axis metrics, it becomes a right axis metric
  * Otherwise it will be added to whatever axis has fewer metrics assigned
* If the left/right axis list is not empty, take the formatter of the first metric in the list as the formatter for the axis
* Assign all metrics of the left/right axis list to the same group

A formatter is compatible if it has the same type, e.g. `number` or `bytes`.

There are other ways of handling this (e.g. displaying multiple axes on a single side if the formatters don't match), but all approaches (including the chosen one) have non-optimal edge cases. These can only be really fixed by providing a more sophisticated way of editing axes which would be out of scope for this PR. We can add this in a later iteration by providing a separate axis management menu in the toolbar.

## Subtle changes

* Prior to this, an axis with non-matching formatters would simply fall back to a plain number formatter. This is confusing in a lot of cases (e.g. https://github.com/elastic/kibana/issues/68260), so this PR changes the logic by just using the first one if the case can't be solved by using two y axes
* There is a separate `yTitle` and `xTitle` argument on the expression to label the axes - so far `xTitle` was only used when there was no column name provided by the data source (never in practice). As we can have multiple y axes now as well, by default the names of the participating columns are used, with the `yTitle` as fallback (like it's happening for `xTitle`).

## Expression integration

Currently y axis series are specified by listing out the columns in the `accessors` argument. To stay compatible with previous versions and not introduce too much new structure, this PR simply adds an `yAxisConfiguration` argument that can be specified multiple times, setting the axis mode to either left, right or auto. As long as the user didn't select anything, no state is written and "auto" mode is assumed. If the user manually picks an axis mode, a `lens_xy_yAxisConfiguration` sub-expression is added:
```
lens_xy_chart {...} yConfig={lens_xy_yConfig forAccessor="a" axisMode="right"}
```

Related to that: In `x-pack/plugins/lens/public/xy_visualization/types.ts` there were expression functions for `YState` and `XConfig` which were unused, I removed them as part of this PR.

`lens_xy_yConfig` can be extended later to add things like metric color overrides.

## `elastic-charts` integration

To use multiple axes, they have to be defined with a `groupId`, then series can be mapped to them by specifying the same `groupId`. Currently there is a single series element in the spec for each layer, but it's possible to have a single layer mapped to different axes. This PR makes this possible, by splitting up each layer for each y accessor, keeping all other properties the same except for the group id which is mapped for each accessor individually. That changes existing specs with more than one metric (e.g. turning one `LineSeries` element into multiple ones).

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56656/)
* Commit: 9fdb807fde00839acb3deedfae648a9c499a0dbe
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56656/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute x-pack-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56656/execution/node/312/log)

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56656/testReport)

<details><summary>Kibana Pipeline / x-pack-intake-agent / X-Pack Jest Tests.x-pack/plugins/lens/public/xy_visualization.xy_expression XYChart component it renders line</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56656/testReport//junit/X-Pack%20Jest%20Tests/x-pack_plugins_lens_public_xy_visualization/Kibana_Pipeline___x_pack_intake_agent___xy_expression_XYChart_component_it_renders_line)


#### Standard Out
```
Failed Tests Reporter:
  - Test has not failed recently on tracked branches


```

#### Stack Trace
```
Error: expect(received).toMatchSnapshot()

Snapshot name: `xy_expression XYChart component it renders line 1`

- Snapshot  - 1
+ Received  + 1

@@ -22,11 +22,11 @@
      id="left"
      key="left"
      position="left"
      showGridLines={false}
      tickFormat={[Function]}
-     title="a"
+     title="a, b"
    />
    <Connect(SpecInstance)
      data={
        Array [
          Object {
    at Object.test (/dev/shm/workspace/kibana/x-pack/plugins/lens/public/xy_visualization/xy_expression.test.tsx:356:25)
    at Promise (/dev/shm/workspace/kibana/node_modules/jest-circus/build/utils.js:198:28)
    at new Promise (<anonymous>)
    at callAsyncCircusFn (/dev/shm/workspace/kibana/node_modules/jest-circus/build/utils.js:162:10)
    at _callCircusTest (/dev/shm/workspace/kibana/node_modules/jest-circus/build/run.js:205:40)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```
</details>

---

<details><summary>Kibana Pipeline / x-pack-intake-agent / X-Pack Jest Tests.x-pack/plugins/lens/public/xy_visualization.xy_expression XYChart component it renders bar</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56656/testReport//junit/X-Pack%20Jest%20Tests/x-pack_plugins_lens_public_xy_visualization/Kibana_Pipeline___x_pack_intake_agent___xy_expression_XYChart_component_it_renders_bar)


#### Standard Out
```
Failed Tests Reporter:
  - Test has not failed recently on tracked branches


```

#### Stack Trace
```
Error: expect(received).toMatchSnapshot()

Snapshot name: `xy_expression XYChart component it renders bar 1`

- Snapshot  - 1
+ Received  + 1

@@ -22,11 +22,11 @@
      id="left"
      key="left"
      position="left"
      showGridLines={false}
      tickFormat={[Function]}
-     title="a"
+     title="a, b"
    />
    <Connect(SpecInstance)
      data={
        Array [
          Object {
    at Object.test (/dev/shm/workspace/kibana/x-pack/plugins/lens/public/xy_visualization/xy_expression.test.tsx:615:25)
    at Promise (/dev/shm/workspace/kibana/node_modules/jest-circus/build/utils.js:198:28)
    at new Promise (<anonymous>)
    at callAsyncCircusFn (/dev/shm/workspace/kibana/node_modules/jest-circus/build/utils.js:162:10)
    at _callCircusTest (/dev/shm/workspace/kibana/node_modules/jest-circus/build/run.js:205:40)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```
</details>

---

<details><summary>Kibana Pipeline / x-pack-intake-agent / X-Pack Jest Tests.x-pack/plugins/lens/public/xy_visualization.xy_expression XYChart component it renders area</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56656/testReport//junit/X-Pack%20Jest%20Tests/x-pack_plugins_lens_public_xy_visualization/Kibana_Pipeline___x_pack_intake_agent___xy_expression_XYChart_component_it_renders_area)


#### Standard Out
```
Failed Tests Reporter:
  - Test has not failed recently on tracked branches


```

#### Stack Trace
```
Error: expect(received).toMatchSnapshot()

Snapshot name: `xy_expression XYChart component it renders area 1`

- Snapshot  - 1
+ Received  + 1

@@ -22,11 +22,11 @@
      id="left"
      key="left"
      position="left"
      showGridLines={false}
      tickFormat={[Function]}
-     title="a"
+     title="a, b"
    />
    <Connect(SpecInstance)
      data={
        Array [
          Object {
    at Object.test (/dev/shm/workspace/kibana/x-pack/plugins/lens/public/xy_visualization/xy_expression.test.tsx:635:25)
    at Promise (/dev/shm/workspace/kibana/node_modules/jest-circus/build/utils.js:198:28)
    at new Promise (<anonymous>)
    at callAsyncCircusFn (/dev/shm/workspace/kibana/node_modules/jest-circus/build/utils.js:162:10)
    at _callCircusTest (/dev/shm/workspace/kibana/node_modules/jest-circus/build/run.js:205:40)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```
</details>

---
and 4 more failures, only showing the first 3.

### Build metrics
> ✅ unchanged

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56656/","number":"56656","commit":"9fdb807fde00839acb3deedfae648a9c499a0dbe"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56749/)
* Commit: 5b48a517707e25f79e319a597a1d55c2b458d6b4

### Build metrics
> ✅ unchanged

### History
* :broken_heart: [Build #56656](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56656/) failed 9fdb807fde00839acb3deedfae648a9c499a0dbe

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56656/","number":"56656","commit":"9fdb807fde00839acb3deedfae648a9c499a0dbe"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56749/","number":"56749","commit":"5b48a517707e25f79e319a597a1d55c2b458d6b4"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/kibana-app (Team:KibanaApp)

mbondyra commented:
@elasticmachine merge upstream

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57037/)
* Commit: 6550d20f32f67cdc0bf3b6f96ef385e6d0156bde
* [Pipeline Steps](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57037/flowGraphTable) (look for red circles / failed steps)
* [Interpreting CI Failures](https://www.elastic.co/guide/en/kibana/current/interpreting-ci-failures.html)

### Failed CI Steps
* [Execute xpack-firefoxSmoke](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57037/execution/node/481/log)
* [Execute kibana-ciGroup9](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57037/execution/node/416/log)
* [Execute kibana-ciGroup8](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57037/execution/node/410/log)
* [Execute kibana-ciGroup7](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57037/execution/node/404/log)
* [Execute kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57037/execution/node/398/log)
* [Execute kibana-ciGroup4](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57037/execution/node/386/log)
* [Execute kibana-ciGroup3](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57037/execution/node/380/log)
* [Execute kibana-ciGroup3](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57037/execution/node/572/log)
* [Execute kibana-ciGroup2](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57037/execution/node/374/log)
* [Execute kibana-firefoxSmoke](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57037/execution/node/362/log)
* [Execute x-pack-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57037/execution/node/316/log)

### Build metrics
> ✅ unchanged

### History
* :green_heart: [Build #56749](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56749/) succeeded 5b48a517707e25f79e319a597a1d55c2b458d6b4
* :broken_heart: [Build #56656](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56656/) failed 9fdb807fde00839acb3deedfae648a9c499a0dbe

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56656/","number":"56656","commit":"9fdb807fde00839acb3deedfae648a9c499a0dbe"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56749/","number":"56749","commit":"5b48a517707e25f79e319a597a1d55c2b458d6b4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57037/","number":"57037","commit":"6550d20f32f67cdc0bf3b6f96ef385e6d0156bde"}]}
PIPELINE-->

flash1293 commented:
@wylieconlon I added this logic, but as you are mentioning it I remembered we chose to only use the first series as label here to avoid overly long axis labels. Rolled back that change

@mbondyra Renamed the expression functions so we can add colors in there as well. Also, I fixed the bug with the unformatted value in the tooltip.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57059/)
* Commit: 970b08a027a8eca966bbdfd63eee6c1805bc2d58

### Build metrics
> ✅ unchanged

### History
* :broken_heart: [Build #57037](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57037/) failed 6550d20f32f67cdc0bf3b6f96ef385e6d0156bde
* :green_heart: [Build #56749](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56749/) succeeded 5b48a517707e25f79e319a597a1d55c2b458d6b4
* :broken_heart: [Build #56656](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56656/) failed 9fdb807fde00839acb3deedfae648a9c499a0dbe

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56656/","number":"56656","commit":"9fdb807fde00839acb3deedfae648a9c499a0dbe"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56749/","number":"56749","commit":"5b48a517707e25f79e319a597a1d55c2b458d6b4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57037/","number":"57037","commit":"6550d20f32f67cdc0bf3b6f96ef385e6d0156bde"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57059/","number":"57059","commit":"970b08a027a8eca966bbdfd63eee6c1805bc2d58"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57104/)
* Commit: 338e0ec57b87196857126c217c6940f944cbd84a

### Build metrics
#### page load asset size
<details><summary>beta</summary><table>
<thead>
<tr>
<th>id</th>
<th><a href="https://github.com/elastic/kibana/commit/338e0ec57b87196857126c217c6940f944cbd84a">value</a></th>
<th>diff</th>
<th><a href="https://github.com/elastic/kibana/commit/3571100bcce63ec3f338a893bd9c549007f7255c">baseline</a></th>
</tr>
</thead>
<tbody>
<tr>
<td>/bundles/app/core/bootstrap.js</td>
<td>-</td>
<td>-13.8KB</td>
<td>13.8KB</td>
</tr>
<tr>
<td>/bundles/core/core.entry.js</td>
<td>-</td>
<td>-1.1MB</td>
<td>1.1MB</td>
</tr>
<tr>
<td>/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.@elastic.js</td>
<td>-</td>
<td>-4.0MB</td>
<td>4.0MB</td>
</tr>
<tr>
<td>/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.css</td>
<td>-</td>
<td>-183.7KB</td>
<td>183.7KB</td>
</tr>
<tr>
<td>/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js</td>
<td>-</td>
<td>-4.6MB</td>
<td>4.6MB</td>
</tr>
<tr>
<td>/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.v7.light.css</td>
<td>-</td>
<td>-511.3KB</td>
<td>511.3KB</td>
</tr>
<tr>
<td>/bundles/light_theme.style.css</td>
<td>-</td>
<td>-95.9KB</td>
<td>95.9KB</td>
</tr>
<tr>
<td>/bundles/plugin/advancedSettings/advancedSettings.plugin.js</td>
<td>-</td>
<td>-941.3KB</td>
<td>941.3KB</td>
</tr>
<tr>
<td>/bundles/plugin/alerts/alerts.plugin.js</td>
<td>-</td>
<td>-89.5KB</td>
<td>89.5KB</td>
</tr>
<tr>
<td>/bundles/plugin/apm/apm.plugin.js</td>
<td>-</td>
<td>-219.2KB</td>
<td>219.2KB</td>
</tr>
<tr>
<td>/bundles/plugin/apmOss/apmOss.plugin.js</td>
<td>-</td>
<td>-4.2KB</td>
<td>4.2KB</td>
</tr>
<tr>
<td>/bundles/plugin/beats_management/beats_management.plugin.js</td>
<td>-</td>
<td>-522.9KB</td>
<td>522.9KB</td>
</tr>
<tr>
<td>/bundles/plugin/bfetch/bfetch.plugin.js</td>
<td>-</td>
<td>-28.7KB</td>
<td>28.7KB</td>
</tr>
<tr>
<td>/bundles/plugin/canvas/canvas.plugin.js</td>
<td>-</td>
<td>-2.0MB</td>
<td>2.0MB</td>
</tr>
<tr>
<td>/bundles/plugin/charts/charts.plugin.js</td>
<td>-</td>
<td>-833.9KB</td>
<td>833.9KB</td>
</tr>
<tr>
<td>/bundles/plugin/cloud/cloud.plugin.js</td>
<td>-</td>
<td>-6.2KB</td>
<td>6.2KB</td>
</tr>
<tr>
<td>/bundles/plugin/console/console.plugin.js</td>
<td>-</td>
<td>-28.0KB</td>
<td>28.0KB</td>
</tr>
<tr>
<td>/bundles/plugin/crossClusterReplication/crossClusterReplication.plugin.js</td>
<td>-</td>
<td>-152.3KB</td>
<td>152.3KB</td>
</tr>
<tr>
<td>/bundles/plugin/dashboard/dashboard.plugin.js</td>
<td>-</td>
<td>-634.2KB</td>
<td>634.2KB</td>
</tr>
<tr>
<td>/bundles/plugin/dashboardEnhanced/dashboardEnhanced.plugin.js</td>
<td>-</td>
<td>-143.5KB</td>
<td>143.5KB</td>
</tr>
<tr>
<td>/bundles/plugin/dashboardMode/dashboardMode.plugin.js</td>
<td>-</td>
<td>-109.3KB</td>
<td>109.3KB</td>
</tr>
<tr>
<td>/bundles/plugin/data/data.plugin.js</td>
<td>-</td>
<td>-1.3MB</td>
<td>1.3MB</td>
</tr>
<tr>
<td>/bundles/plugin/dataEnhanced/dataEnhanced.plugin.js</td>
<td>-</td>
<td>-138.1KB</td>
<td>138.1KB</td>
</tr>
<tr>
<td>/bundles/plugin/devTools/devTools.plugin.js</td>
<td>-</td>
<td>-123.9KB</td>
<td>123.9KB</td>
</tr>
<tr>
<td>/bundles/plugin/discover/discover.plugin.js</td>
<td>-</td>
<td>-221.5KB</td>
<td>221.5KB</td>
</tr>
<tr>
<td>/bundles/plugin/discoverEnhanced/discoverEnhanced.plugin.js</td>
<td>-</td>
<td>-28.1KB</td>
<td>28.1KB</td>
</tr>
<tr>
<td>/bundles/plugin/embeddable/embeddable.plugin.js</td>
<td>-</td>
<td>-390.9KB</td>
<td>390.9KB</td>
</tr>
<tr>
<td>/bundles/plugin/embeddableEnhanced/embeddableEnhanced.plugin.js</td>
<td>-</td>
<td>-26.3KB</td>
<td>26.3KB</td>
</tr>
<tr>
<td>/bundles/plugin/esUiShared/esUiShared.plugin.js</td>
<td>-</td>
<td>-927.8KB</td>
<td>927.8KB</td>
</tr>
<tr>
<td>/bundles/plugin/expressions/expressions.plugin.js</td>
<td>-</td>
<td>-314.9KB</td>
<td>314.9KB</td>
</tr>
<tr>
<td>/bundles/plugin/features/features.plugin.js</td>
<td>-</td>
<td>-13.4KB</td>
<td>13.4KB</td>
</tr>
<tr>
<td>/bundles/plugin/fileUpload/fileUpload.plugin.js</td>
<td>-</td>
<td>-688.0KB</td>
<td>688.0KB</td>
</tr>
<tr>
<td>/bundles/plugin/globalSearch/globalSearch.plugin.js</td>
<td>-</td>
<td>-29.0KB</td>
<td>29.0KB</td>
</tr>
<tr>
<td>/bundles/plugin/graph/graph.plugin.js</td>
<td>-</td>
<td>-15.3KB</td>
<td>15.3KB</td>
</tr>
<tr>
<td>/bundles/plugin/grokdebugger/grokdebugger.plugin.js</td>
<td>-</td>
<td>-11.2KB</td>
<td>11.2KB</td>
</tr>
<tr>
<td>/bundles/plugin/home/home.plugin.js</td>
<td>-</td>
<td>-24.6KB</td>
<td>24.6KB</td>
</tr>
<tr>
<td>/bundles/plugin/indexLifecycleManagement/indexLifecycleManagement.plugin.js</td>
<td>-</td>
<td>-184.2KB</td>
<td>184.2KB</td>
</tr>
<tr>
<td>/bundles/plugin/indexManagement/indexManagement.plugin.js</td>
<td>-</td>
<td>-201.7KB</td>
<td>201.7KB</td>
</tr>
<tr>
<td>/bundles/plugin/indexPatternManagement/indexPatternManagement.plugin.js</td>
<td>-</td>
<td>-137.3KB</td>
<td>137.3KB</td>
</tr>
<tr>
<td>/bundles/plugin/infra/infra.plugin.js</td>
<td>-</td>
<td>-92.9KB</td>
<td>92.9KB</td>
</tr>
<tr>
<td>/bundles/plugin/ingestPipelines/ingestPipelines.plugin.js</td>
<td>-</td>
<td>-30.8KB</td>
<td>30.8KB</td>
</tr>
<tr>
<td>/bundles/plugin/inputControlVis/inputControlVis.plugin.js</td>
<td>-</td>
<td>-258.9KB</td>
<td>258.9KB</td>
</tr>
<tr>
<td>/bundles/plugin/inspector/inspector.plugin.js</td>
<td>-</td>
<td>-238.3KB</td>
<td>238.3KB</td>
</tr>
<tr>
<td>/bundles/plugin/kibanaLegacy/kibanaLegacy.plugin.js</td>
<td>-</td>
<td>-174.7KB</td>
<td>174.7KB</td>
</tr>
<tr>
<td>/bundles/plugin/kibanaReact/kibanaReact.plugin.js</td>
<td>-</td>
<td>-575.3KB</td>
<td>575.3KB</td>
</tr>
<tr>
<td>/bundles/plugin/kibanaUtils/kibanaUtils.plugin.js</td>
<td>-</td>
<td>-434.4KB</td>
<td>434.4KB</td>
</tr>
<tr>
<td>/bundles/plugin/lens/lens.plugin.js</td>
<td>-</td>
<td>-758.3KB</td>
<td>758.3KB</td>
</tr>
<tr>
<td>/bundles/plugin/licenseManagement/licenseManagement.plugin.js</td>
<td>-</td>
<td>-25.8KB</td>
<td>25.8KB</td>
</tr>
<tr>
<td>/bundles/plugin/licensing/licensing.plugin.js</td>
<td>-</td>
<td>-21.6KB</td>
<td>21.6KB</td>
</tr>
<tr>
<td>/bundles/plugin/logstash/logstash.plugin.js</td>
<td>-</td>
<td>-141.8KB</td>
<td>141.8KB</td>
</tr>
<tr>
<td>/bundles/plugin/management/management.plugin.js</td>
<td>-</td>
<td>-25.9KB</td>
<td>25.9KB</td>
</tr>
<tr>
<td>/bundles/plugin/maps/maps.plugin.js</td>
<td>-</td>
<td>-220.6KB</td>
<td>220.6KB</td>
</tr>
<tr>
<td>/bundles/plugin/mapsLegacy/mapsLegacy.plugin.js</td>
<td>-</td>
<td>-1.0MB</td>
<td>1.0MB</td>
</tr>
<tr>
<td>/bundles/plugin/mapsLegacyLicensing/mapsLegacyLicensing.plugin.js</td>
<td>-</td>
<td>-3.9KB</td>
<td>3.9KB</td>
</tr>
<tr>
<td>/bundles/plugin/ml/ml.plugin.js</td>
<td>-</td>
<td>-1.3MB</td>
<td>1.3MB</td>
</tr>
<tr>
<td>/bundles/plugin/monitoring/monitoring.plugin.js</td>
<td>-</td>
<td>-22.8KB</td>
<td>22.8KB</td>
</tr>
<tr>
<td>/bundles/plugin/navigation/navigation.plugin.js</td>
<td>-</td>
<td>-122.2KB</td>
<td>122.2KB</td>
</tr>
<tr>
<td>/bundles/plugin/newsfeed/newsfeed.plugin.js</td>
<td>-</td>
<td>-22.7KB</td>
<td>22.7KB</td>
</tr>
<tr>
<td>/bundles/plugin/observability/observability.plugin.js</td>
<td>-</td>
<td>-103.1KB</td>
<td>103.1KB</td>
</tr>
<tr>
<td>/bundles/plugin/painlessLab/painlessLab.plugin.js</td>
<td>-</td>
<td>-158.8KB</td>
<td>158.8KB</td>
</tr>
<tr>
<td>/bundles/plugin/regionMap/regionMap.plugin.js</td>
<td>-</td>
<td>-791.7KB</td>
<td>791.7KB</td>
</tr>
<tr>
<td>/bundles/plugin/remoteClusters/remoteClusters.plugin.js</td>
<td>-</td>
<td>-35.2KB</td>
<td>35.2KB</td>
</tr>
<tr>
<td>/bundles/plugin/reporting/reporting.plugin.js</td>
<td>-</td>
<td>-257.7KB</td>
<td>257.7KB</td>
</tr>
<tr>
<td>/bundles/plugin/rollup/rollup.plugin.js</td>
<td>-</td>
<td>-186.0KB</td>
<td>186.0KB</td>
</tr>
<tr>
<td>/bundles/plugin/savedObjects/savedObjects.plugin.js</td>
<td>-</td>
<td>-196.2KB</td>
<td>196.2KB</td>
</tr>
<tr>
<td>/bundles/plugin/savedObjectsManagement/savedObjectsManagement.plugin.js</td>
<td>-</td>
<td>-176.6KB</td>
<td>176.6KB</td>
</tr>
<tr>
<td>/bundles/plugin/searchprofiler/searchprofiler.plugin.js</td>
<td>-</td>
<td>-42.8KB</td>
<td>42.8KB</td>
</tr>
<tr>
<td>/bundles/plugin/security/security.plugin.js</td>
<td>-</td>
<td>-248.0KB</td>
<td>248.0KB</td>
</tr>
<tr>
<td>/bundles/plugin/share/share.plugin.js</td>
<td>-</td>
<td>-80.5KB</td>
<td>80.5KB</td>
</tr>
<tr>
<td>/bundles/plugin/snapshotRestore/snapshotRestore.plugin.js</td>
<td>-</td>
<td>-57.5KB</td>
<td>57.5KB</td>
</tr>
<tr>
<td>/bundles/plugin/spaces/spaces.plugin.js</td>
<td>-</td>
<td>-367.2KB</td>
<td>367.2KB</td>
</tr>
<tr>
<td>/bundles/plugin/statusPage/statusPage.plugin.js</td>
<td>-</td>
<td>-3.6KB</td>
<td>3.6KB</td>
</tr>
<tr>
<td>/bundles/plugin/telemetry/telemetry.plugin.js</td>
<td>-</td>
<td>-77.0KB</td>
<td>77.0KB</td>
</tr>
<tr>
<td>/bundles/plugin/telemetryManagementSection/telemetryManagementSection.plugin.js</td>
<td>-</td>
<td>-27.5KB</td>
<td>27.5KB</td>
</tr>
<tr>
<td>/bundles/plugin/tileMap/tileMap.plugin.js</td>
<td>-</td>
<td>-807.1KB</td>
<td>807.1KB</td>
</tr>
<tr>
<td>/bundles/plugin/transform/transform.plugin.js</td>
<td>-</td>
<td>-21.6KB</td>
<td>21.6KB</td>
</tr>
<tr>
<td>/bundles/plugin/triggers_actions_ui/triggers_actions_ui.plugin.js</td>
<td>-</td>
<td>-103.0KB</td>
<td>103.0KB</td>
</tr>
<tr>
<td>/bundles/plugin/uiActions/uiActions.plugin.js</td>
<td>-</td>
<td>-162.5KB</td>
<td>162.5KB</td>
</tr>
<tr>
<td>/bundles/plugin/uiActionsEnhanced/uiActionsEnhanced.plugin.js</td>
<td>-</td>
<td>-130.7KB</td>
<td>130.7KB</td>
</tr>
<tr>
<td>/bundles/plugin/upgradeAssistant/upgradeAssistant.plugin.js</td>
<td>-</td>
<td>-46.6KB</td>
<td>46.6KB</td>
</tr>
<tr>
<td>/bundles/plugin/uptime/uptime.plugin.js</td>
<td>-</td>
<td>-327.3KB</td>
<td>327.3KB</td>
</tr>
<tr>
<td>/bundles/plugin/usageCollection/usageCollection.plugin.js</td>
<td>-</td>
<td>-24.4KB</td>
<td>24.4KB</td>
</tr>
<tr>
<td>/bundles/plugin/visTypeMarkdown/visTypeMarkdown.plugin.js</td>
<td>-</td>
<td>-512.5KB</td>
<td>512.5KB</td>
</tr>
<tr>
<td>/bundles/plugin/visTypeMetric/visTypeMetric.plugin.js</td>
<td>-</td>
<td>-536.7KB</td>
<td>536.7KB</td>
</tr>
<tr>
<td>/bundles/plugin/visTypeTable/visTypeTable.plugin.js</td>
<td>-</td>
<td>-563.0KB</td>
<td>563.0KB</td>
</tr>
<tr>
<td>/bundles/plugin/visTypeTagcloud/visTypeTagcloud.plugin.js</td>
<td>-</td>
<td>-797.1KB</td>
<td>797.1KB</td>
</tr>
<tr>
<td>/bundles/plugin/visTypeTimelion/visTypeTimelion.plugin.js</td>
<td>-</td>
<td>-670.7KB</td>
<td>670.7KB</td>
</tr>
<tr>
<td>/bundles/plugin/visTypeTimeseries/visTypeTimeseries.plugin.js</td>
<td>-</td>
<td>-212.2KB</td>
<td>212.2KB</td>
</tr>
<tr>
<td>/bundles/plugin/visTypeVega/visTypeVega.plugin.js</td>
<td>-</td>
<td>-580.3KB</td>
<td>580.3KB</td>
</tr>
<tr>
<td>/bundles/plugin/visTypeVislib/visTypeVislib.plugin.js</td>
<td>-</td>
<td>-1.2MB</td>
<td>1.2MB</td>
</tr>
<tr>
<td>/bundles/plugin/visualizations/visualizations.plugin.js</td>
<td>-</td>
<td>-343.4KB</td>
<td>343.4KB</td>
</tr>
<tr>
<td>/bundles/plugin/visualize/visualize.plugin.js</td>
<td>-</td>
<td>-18.2KB</td>
<td>18.2KB</td>
</tr>
<tr>
<td>/bundles/plugin/watcher/watcher.plugin.js</td>
<td>-</td>
<td>-35.2KB</td>
<td>35.2KB</td>
</tr>
<tr>
<td>/node_modules/@kbn/ui-framework/dist/kui_light.css</td>
<td>-</td>
<td>-62.1KB</td>
<td>62.1KB</td>
</tr>
<tr>
<td>/translations/en.json</td>
<td>-</td>
<td>-29.0B</td>
<td>29.0B</td>
</tr>
<tr>
<td>/ui/fonts/inter_ui/Inter-UI-Medium.woff2</td>
<td>-</td>
<td>-92.2KB</td>
<td>92.2KB</td>
</tr>
<tr>
<td>/ui/fonts/inter_ui/Inter-UI-Regular.woff2</td>
<td>-</td>
<td>-86.4KB</td>
<td>86.4KB</td>
</tr>
<tr>
<td>total</td>
<td>-</td>
<td>-36.2MB</td>
<td>-</td>
</tr>
</tbody>
</table>
</details>

### History
* :green_heart: [Build #57059](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57059/) succeeded 970b08a027a8eca966bbdfd63eee6c1805bc2d58
* :broken_heart: [Build #57037](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57037/) failed 6550d20f32f67cdc0bf3b6f96ef385e6d0156bde
* :green_heart: [Build #56749](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56749/) succeeded 5b48a517707e25f79e319a597a1d55c2b458d6b4
* :broken_heart: [Build #56656](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56656/) failed 9fdb807fde00839acb3deedfae648a9c499a0dbe

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56656/","number":"56656","commit":"9fdb807fde00839acb3deedfae648a9c499a0dbe"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56749/","number":"56749","commit":"5b48a517707e25f79e319a597a1d55c2b458d6b4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57037/","number":"57037","commit":"6550d20f32f67cdc0bf3b6f96ef385e6d0156bde"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57059/","number":"57059","commit":"970b08a027a8eca966bbdfd63eee6c1805bc2d58"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57104/","number":"57104","commit":"338e0ec57b87196857126c217c6940f944cbd84a"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57183/)
* Commit: c8b8ec8f122def7f72bd7ab19a791221b30ef6e9

### Build metrics
> ✅ unchanged

### History
* :green_heart: [Build #57104](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57104/) succeeded 338e0ec57b87196857126c217c6940f944cbd84a
* :green_heart: [Build #57059](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57059/) succeeded 970b08a027a8eca966bbdfd63eee6c1805bc2d58
* :broken_heart: [Build #57037](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57037/) failed 6550d20f32f67cdc0bf3b6f96ef385e6d0156bde
* :green_heart: [Build #56749](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56749/) succeeded 5b48a517707e25f79e319a597a1d55c2b458d6b4
* :broken_heart: [Build #56656](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56656/) failed 9fdb807fde00839acb3deedfae648a9c499a0dbe

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56656/","number":"56656","commit":"9fdb807fde00839acb3deedfae648a9c499a0dbe"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/56749/","number":"56749","commit":"5b48a517707e25f79e319a597a1d55c2b458d6b4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57037/","number":"57037","commit":"6550d20f32f67cdc0bf3b6f96ef385e6d0156bde"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57059/","number":"57059","commit":"970b08a027a8eca966bbdfd63eee6c1805bc2d58"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57104/","number":"57104","commit":"338e0ec57b87196857126c217c6940f944cbd84a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/57183/","number":"57183","commit":"c8b8ec8f122def7f72bd7ab19a791221b30ef6e9"}]}
PIPELINE-->

wylieconlon commented on x-pack/plugins/lens/public/xy_visualization/xy_expression.tsx:
```diff
@@ -276,6 +276,9 @@ export function XYChart({
         showLegendExtra={false}
         theme={chartTheme}
+        tooltip={{
+          headerFormatter: (d) => xAxisFormatter.convert(d.value),
```
This does of course fail if there are non-matching types on the X axis, but I think we can't fix that case.
